auth/cache_test.go

```diff
@@ -0,0 +1,366 @@
+		AccessToken:  tok2.AccessToken,
+		RefreshToken: tok2.RefreshToken,
+		Expiry:       tok2.Expiry,
+		MaxAge:       MaxAge,
+	}
+	assert.Equal(expectedCache.AccessToken, cacheFromJSON.AccessToken)
+	assert.Equal(expectedCache.RefreshToken, cacheFromJSON.RefreshToken)
+	assert.Equal(expectedCache.Expiry.Format("2006-01-02T15:04:05 -07:00:00"), cacheFromJSON.Expiry.Format("2006-01-02T15:04:05 -07:00:00"))
+	assert.Equal(expectedCache.MaxAge.String(), cacheFromJSON.MaxAge.String())
+	require.NoError(destroy())
+}
+
+func TestRead_InvalidFormat(t *testing.T) {
+	require := requirepkg.New(t)
+	assert := assertpkg.New(t)
+
+	credentialDirectory, credentialPath, err := setup()
+	require.NoError(err)
+	require.NotNil(credentialDirectory)
+
+	err = os.MkdirAll(credentialDirectory, directoryPermissions)
+	if err != nil {
+		fmt.Printf("not able to make test directory :%v", err)
+	}
+	require.NoError(err)
+
+	type redHerring struct {
+		AccessToken  string
+		anotherField string
+		lastField    string
+	}
+
+	randomData := redHerring{
+		AccessToken:  "TopSecret!",
+		anotherField: "SoRefreshing:)",
+		lastField:    "field",
+	}
+
+	randomDataJSON, err := json.Marshal(randomData)
+	require.NoError(err)
+
+	err = os.WriteFile(credentialPath, randomDataJSON, directoryPermissions)
+	require.NoError(err)
+
+	_, err = Read()
+	require.Error(err)
+	assert.EqualError(err, "bad format: failed to get cache access token")
+	require.NoError(destroy())
+}
+
+func TestRead_ValidFormat(t *testing.T) {
+	require := requirepkg.New(t)
+	assert := assertpkg.New(t)
+
+	credentialDirectory, _, err := setup()
+	require.NoError(err)
+	require.NotNil(credentialDirectory)
+
+	now := time.Now()
+	tok := oauth2.Token{
+		AccessToken:  "TopSecret!",
+		RefreshToken: "SoRefreshing:)",
+		Expiry:       now,
+	}
+
+	assert.NoError(Write(tok))
+
+	cachePointer, err := Read()
+	require.NoError(err)
+
+	expectedCache := Cache{
+		AccessToken:  tok.AccessToken,
+		RefreshToken: tok.RefreshToken,
+		Expiry:       tok.Expiry,
+		MaxAge:       MaxAge,
+	}
+
+	assert.Equal(expectedCache.AccessToken, cachePointer.AccessToken)
+	assert.Equal(expectedCache.RefreshToken, cachePointer.RefreshToken)
+	assert.Equal(expectedCache.Expiry.Format("2006-01-02T15:04:05 -07:00:00"), cachePointer.Expiry.Format("2006-01-02T15:04:05 -07:00:00"))
+	assert.Equal(expectedCache.MaxAge.String(), cachePointer.MaxAge.String())
+	require.NoError(destroy())
+}
+
+func TestGetCredentialPaths_ReturnsPaths(t *testing.T) {
+	os.Setenv(envVarCacheTestMode, "true")
+
+	assert := assertpkg.New(t)
+
+	credentialPath, credentialDir, err := getCredentialPaths()
+	assert.NoError(err)
+
+	userHome, err := os.UserHomeDir()
+	assert.NoError(err)
+	expectedDirectory := filepath.Join(userHome, testDirectory)
+	expectedPath := filepath.Join(userHome, testDirectory, fileName)
+
+	assert.Equal(expectedDirectory, credentialDir)
+	assert.Equal(expectedPath, credentialPath)
+}
+
+func TestJsonToToken_BadJSONThrowsError(t *testing.T) {
```

Comment:
For consistency with our other test names. We should add a Valid format test too.
```suggestion
func TestJsonToToken_InvalidFormat(t *testing.T) {
```